### PR TITLE
Removes the input witness struct of ASForR1CSNark

### PR DESCRIPTION
**Pull request by Will-Lin4**
The input witness of ASForR1CSNark is the SecondRoundMessage of R1CSNark. There is no need for a wrapper around it, so the commit removes the struct and uses SecondRoundMessage directly.

### CODE REVIEW

**Comment by Will-Lin4** [comment]
The CI is failing because a nightly test is detecting an unused variable. This issue is fixed in #24. After that PR is merged, we can rerun the tests.

File: src/r1cs_nark_as/data_structures.rs [hunk 1]
```diff
@@ -101,24 +99,6 @@ where
...
-    pub second_round_message: SecondRoundMessage<F>,
-}
-
-impl<F: Field> InputWitness<F> {
-    pub(crate) fn zero(witness_len: usize) -> Self {
-        Self {
-            second_round_message: SecondRoundMessage::zero(witness_len),
-        }
-    }
-}
```
Comment (Pratyush) [inline src/r1cs_nark_as/data_structures.rs:120]: Maybe we can replace this with `pub type InputWitness<F> = SecondRoundMessage<F>`?

Comment (Pratyush) [reply src/r1cs_nark_as/data_structures.rs:120]: That way logically the type names in the rest of this PR don't have to change from `InputWitness` to `SecondRoundMessage`, while removing the layer of indirection from before.

**Review by Pratyush — APPROVED** [review]
LGTM modulo the nit

**Pull request merged**